From bfde6df388f2373ba93157ce565f21635aa394a4 Mon Sep 17 00:00:00 2001
From: Bin Han <bin.b.han@intel.com>
Date: Mon, 13 Oct 2014 10:45:52 +0800
Subject: [PATCH 153/164] atomisp2: the last MMU level 1 page table entry is
 never used

BZ: 226217

Due to the off-by-1 error, the last L1 page table entry is never used, thus
fix it.

Change-Id: Idde42eb80b7caa4e4f4bf489ba064de159f5ff05
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-8929
Signed-off-by: Bin Han <bin.b.han@intel.com>
Reviewed-on: https://android.intel.com:443/287104
Reviewed-on: https://icggerrit.ir.intel.com/29112
Approver: Wei Tao <wei.tao@intel.com>
Build: Wei Tao <wei.tao@intel.com>
Reviewed-by: Wei Tao <wei.tao@intel.com>
Maintainer: Wei Tao <wei.tao@intel.com>
---
 .../pci/atomisp2/atomisp_driver/mmu/isp_mmu.c      |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/mmu/isp_mmu.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/mmu/isp_mmu.c
index a8dc94f..c9c88aa 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/mmu/isp_mmu.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/mmu/isp_mmu.c
@@ -322,7 +322,7 @@ static int mmu_l1_map(struct isp_mmu *mmu, phys_addr_t l1_pt,
 
 			return -EINVAL;
 		}
-	} while (ptr < end && idx < ISP_L1PT_PTES - 1);
+	} while (ptr < end && idx < ISP_L1PT_PTES);
 
 	return 0;
 }
@@ -463,7 +463,7 @@ static void mmu_l1_unmap(struct isp_mmu *mmu, phys_addr_t l1_pt,
 		 * need to invalidate and free this PT.
 		 */
 		/*      atomisp_set_pte(l1_pt, idx, NULL_PTE); */
-	} while (ptr < end && idx < ISP_L1PT_PTES - 1);
+	} while (ptr < end && idx < ISP_L1PT_PTES);
 }
 
 /*
-- 
1.7.9.5

